.travis.yml

```diff
@@ -20,8 +20,15 @@ addons:
 
 before_script:
 - |
-  pip install 'travis-cargo<0.2' --user &&
-  export PATH=$HOME/.local/bin:/usr/local/bin:$PATH
+  if [ $TRAVIS_OS_NAME = 'osx' ]; then
+    brew install python3
+    virtualenv env -p python3
+    source env/bin/activate
+    pip install 'travis-cargo<0.2'
+  else
+    pip install 'travis-cargo<0.2' --user
+    export PATH="$(python -m site --user-base)/bin:$PATH"
```

Comment:
One other thing worth pointing out is that the `before_script` was previously
```
pip install 'travis-cargo<0.2' --user && export PATH=$HOME/.local/bin:/usr/local/bin:$PATH
```
This ensured that the `export` did not run unless the `pip install` succeeded. The changes in this PR break that, and if the `pip install` fails, I think (disclosure: I don't have much experience with Travis CI) that Travis CI will keep going after `before_script` until something fails due to the absence of `travis-cargo`. That seems to be the implied consequence in travis-ci/travis-ci#1066.

There's a similar situation with the macOS branch of the condition in these changes, which has more opportunities to fail.

I believe the right action here is to chain the commands in both branches of the `if` statement with `&&`. See, for example, rust-lang/rust#12513, where this was done.

Reply:
If you wanted to experiment, you could also try putting a `set -e` at the beginning of the script and a `set +e` at the end. You would do that instead of the `&&` chaining, which would be a burden if the script was really long. If you don't want to experiment, the chaining is fine, though, since it's not that long, and it's the convention elsewhere in the file already.